### PR TITLE
[BEAM-1316] start bundle should not output any elements

### PR DESCRIPTION
**Pull request by sb2nov**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

R: @chamikaramj @robertwb PTAL

### CODE REVIEW

File: sdks/python/apache_beam/runners/common.py [hunk 1]
```diff
@@ -482,6 +470,50 @@ def process_outputs(self, windowed_input_element, results):
...
+        windowed_value = result
+      elif isinstance(result, TimestampedValue):
+        value = result.value
+        timestamp = result.timestamp
+        assign_context = NoContext(value, timestamp)
+        windowed_value = WindowedValue(
+            value, timestamp, self.window_fn.assign(assign_context))
+      else:
+        value = result
+        timestamp = -1
```
Comment (aaltay) [inline sdks/python/apache_beam/runners/common.py]: Should we convert this case to assert per discussion in https://issues.apache.org/jira/browse/BEAM-1283

File: sdks/python/apache_beam/transforms/ptransform_test.py [hunk 2]
```diff
@@ -274,31 +285,29 @@ def incorrect_par_do_fn(x):
...
-        equal_to(['start', 'finish'])(list(set(actual)))
-        equal_to([actual.count('start')])([actual.count('finish')])
+        equal_to(['finish'])(list(set(actual)))
+        equal_to([1])([actual.count('finish')])
       return match
 
     assert_that(result, matcher())
     pipeline.run()
 
+  @mock.patch.object(MyDoFn, 'start_bundle')
```
Comment (aaltay) [inline sdks/python/apache_beam/transforms/ptransform_test.py]: Should there be a test for the error case? (returning from start_bundle)

File: sdks/python/apache_beam/runners/common.py (hunk 1 above)
Comment (sb2nov) [inline sdks/python/apache_beam/runners/common.py]: As that is a breaking change I wanted to isolate that in another PR

File: sdks/python/apache_beam/transforms/ptransform_test.py (hunk 2 above)
Comment (sb2nov) [inline sdks/python/apache_beam/transforms/ptransform_test.py]: Good point. Will add.

Comment (sb2nov) [inline sdks/python/apache_beam/transforms/ptransform_test.py]: Done.

**Comment by sb2nov** [comment]
@aaltay PTAL again

File: sdks/python/apache_beam/runners/common.py (hunk 1 above)
Comment (aaltay) [inline sdks/python/apache_beam/runners/common.py]: OK. (Note that this is also a breaking change.)

**Comment by sb2nov** [comment]
Run Python PostCommit

**Comment by aaltay** [comment]
I manually triggered https://builds.apache.org/view/Beam/job/beam_PostCommit_Python_Verify/2002/ but it has a queue ahead of it.

**Comment by aaltay** [comment]
post commit job was successful, I will merge.

**Pull request closed without merging**